src/multidim/intervalbox.jl

```diff
@@ -109,3 +109,28 @@ IntervalBox(x::Interval, n::Int) = IntervalBox(x, Val(n))
 dot(x::IntervalBox, y::IntervalBox) = dot(x.v, y.v)
 
 Base.:(==)(x::IntervalBox, y::IntervalBox) = x.v == y.v
+
+
+"""
+    mince(x::IntervalBox, n)
+
+Splits `x` in `n` intervals in each dimension of the same diameter. These
+intervals are combined in all possible `IntervalBox`-es, which are returned
+as a vector.
+"""
+@generated function mince(x::IntervalBox{N,T}, n) where {N,T}
+    quote
+        nodes_matrix = Array{Interval{T},2}(undef, n, N)
+        for i in 1:N
+            nodes_matrix[1:n,i] .= mince(x[i], n)
+        end
+
+        nodes = IntervalBox{$N,T}[]
+        Base.Cartesian.@nloops $N i _->(1:n) begin
```

Comment:
I'm pretty sure there's an easier way to do this.

Reply:
e.g. using something in https://julialang.org/blog/2016/02/iteration/ or in the `Iterators` module.

Reply:
I'll check the link; in any case, my naive checks yield a nice performance. This is also a good time to see if iteration is robust for `IntervalBox`-es.

Reply:
I've implemented something, which works, with `CartesianIndices`, but the performance is worst. I think we can leave like as it is for the time being.